ludwig/contribs/mlflow/__init__.py

```diff
@@ -80,14 +84,16 @@ def on_train_init(self, base_config, experiment_name, output_directory, resume_d
                 run_name = os.path.basename(output_directory)
                 self.run = mlflow.start_run(experiment_id=self.experiment_id, run_name=run_name)
 
-        mlflow.log_dict(to_json_dict(base_config), "config.yaml")
+        if self.log_artifacts:
```

Comment:
I don't think this is an artifact. Can you clarify this one?

Reply:
This logs the base config as an artifact, discussed more in the docs for [mlflow.log_dict](https://www.mlflow.org/docs/latest/python_api/mlflow.html#mlflow.log_dict). The dictionary gets logged as a yaml file like in the screenshot below:
<img width="1376" alt="Screen Shot 2022-07-27 at 3 36 14 PM" src="https://user-images.githubusercontent.com/5787923/181385451-7aacd803-f53f-4358-be60-c75ec678e6a2.png">

We could still want to log this dictionary as an artifact even when artifact logging for other files is turned off. In that case however, the artifact locations, etc. will all be managed by mlflow.

Reply:
For something like the config it seems reasonable to log regardless.

Reply:
Ah, I see. I agree with disabling it, then, because otherwise mlflow will just write it to the local (ephemeral) disk.